src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -4953,6 +4946,51 @@ class C
             //     public required int Test { get; set; }
             Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "Test").WithArguments("C", "Test").WithLocation(4, 25)
         );
+    }
+
+    [Theory, CombinatorialData]
+    public void FirstAccessOfIsRequiredDoesNotMatter(bool accessAttributesFirst)
+    {
+        // Accessing attributes will populate IsRequired if it's not already populated, so we want to test both codepaths explicitly.
+        var comp = CreateCompilationWithRequiredMembers("""
+            public class C
+            {
+                public required int Field;
+                public required int Property { get; set; }
+            }
+            """);
 
+        CompileAndVerify(comp, symbolValidator: module =>
+        {
+            var c = module.ContainingAssembly.GetTypeByMetadataName("C");
+            AssertEx.NotNull(c);
+            FieldSymbol fieldSymbol = c.GetMember<FieldSymbol>("Field");
+            PropertySymbol propertySymbol = c.GetMember<PropertySymbol>("Property");
+
+            if (accessAttributesFirst)
+            {
+                assertAttributesEmpty();
+                assertIsRequired();
+            }
+            else
+            {
+                assertIsRequired();
+                assertAttributesEmpty();
+            }
+
+            void assertIsRequired()
+            {
+                Assert.True(c.HasDeclaredRequiredMembers);
```

Comment:
>True [](http://example.com/codeflow?start=23&length=4)

It looks like we are only testing scenario when the attribute is supposed to be present in metadata. I think we should also test a scenario when the attribute is not present.

Reply:
It looks like negative scenario for types isn't tested.